src/evalcontext.cc

```diff
@@ -35,19 +35,36 @@ ValuePtr EvalContext::getArgValue(size_t i, const Context *ctx) const
   Resolves arguments specified by evalctx, using args to lookup positional arguments.
   Returns an AssignmentMap (string -> Expression*)
 */
-AssignmentMap EvalContext::resolveArguments(const AssignmentList &args) const
+AssignmentMap EvalContext::resolveArguments(const AssignmentList &args, const AssignmentList &optargs, bool silent) const
 {
   AssignmentMap resolvedArgs;
   size_t posarg = 0;
+  bool tooManyWarned=false;
   // Iterate over positional args
   for (size_t i=0; i<this->numArgs(); i++) {
     const auto &name = this->getArgName(i); // name is optional
     const auto expr = this->getArgs()[i].expr.get();
     if (!name.empty()) {
+      if(name.at(0)!='$' && !silent){
+        bool found=false;
+        for(auto const& arg: args) {
+          if(arg.name == name) found=true;
+        }
+        for(auto const& arg: optargs) {
+          if(arg.name == name) found=true;
+        }
+        if(!found){
+          PRINTB("WARNING: variable %s not specified as parameter, %s", name % this->loc.toRelativeString(this->documentPath()));
+        }
+      }
       resolvedArgs[name] = expr;
     }
     // If positional, find name of arg with this position
     else if (posarg < args.size()) resolvedArgs[args[posarg++].name] = expr;
+    else if (!tooManyWarned){
+      PRINTB("WARNING: Too many unnamed arguments supplied, %s", this->loc.toRelativeString(this->documentPath()));
+      tooManyWarned=true;
```

Comment:
hmm... as the bool is now call silent, this should also be conditional.
(despite the fact that supplying too many positional arguments is a separate issue from using unspecified parameters)